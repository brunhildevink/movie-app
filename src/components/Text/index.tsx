import styled from 'styled-components'

import { fonts, typography } from '../../styles'

const sharedCss = `
font-family: ${fonts};
font-size: ${typography.fontSizeText}px;
line-height: ${typography.lineHeightText};

&:first-child {
  margin-top: 0;
}

&:last-child {
  margin-bottom: 0;
}
`

const Bold = styled.p`
  font-weight: ${typography.fontWeightBold};

  ${sharedCss}
`

const Light = styled.p`
  font-weight: ${typography.fontWeightLight};

  ${sharedCss}
`

const Medium = styled.p`
  font-weight: ${typography.fontWeightMedium};

  ${sharedCss}
`

const Regular = styled.p`
  font-weight: ${typography.fontWeightRegular};

  ${sharedCss}
`

export default { Bold, Light, Medium, Regular }
