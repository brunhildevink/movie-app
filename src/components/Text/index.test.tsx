import React from 'react'
import renderer from 'react-test-renderer'
import Text from '.'

test('Text bold renders correctly', () => {
  const tree = renderer.create(<Text.Bold />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Text light renders correctly', () => {
  const tree = renderer.create(<Text.Light />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Text medium renders correctly', () => {
  const tree = renderer.create(<Text.Medium />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Text regular renders correctly', () => {
  const tree = renderer.create(<Text.Regular />).toJSON()
  expect(tree).toMatchSnapshot()
})
