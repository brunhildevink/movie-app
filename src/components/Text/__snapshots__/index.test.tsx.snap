// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text bold renders correctly 1`] = `
<p
  className="sc-bczRLJ YRWNg"
/>
`;

exports[`Text light renders correctly 1`] = `
<p
  className="sc-gsnTZi bMDWaK"
/>
`;

exports[`Text medium renders correctly 1`] = `
<p
  className="sc-dkzDqf cnoQUb"
/>
`;

exports[`Text regular renders correctly 1`] = `
<p
  className="sc-hKMtZM xSPEl"
/>
`;
